pkg/serving/v1alpha1/client_test.go

```diff
@@ -285,6 +285,39 @@ func TestListRevisionForService(t *testing.T) {
 		})
 }
 
+func TestGetRoute(t *testing.T) {
+	serving, client := setup()
+	routeName := "test-route"
+
+	serving.AddReactor("get", "routes",
+		func(a client_testing.Action) (bool, runtime.Object, error) {
+			route := newRoute(routeName)
+			name := a.(client_testing.GetAction).GetName()
+			// Sanity check
+			assert.Assert(t, name != "")
+			assert.Equal(t, testNamespace, a.GetNamespace())
+			if name == routeName {
+				return true, route, nil
+			}
+			return true, nil, errors.NewNotFound(v1alpha1.Resource("route"), name)
+		})
+
+	t.Run("get known route by name returns route", func(t *testing.T) {
+		route, err := client.GetRoute(routeName)
+		assert.NilError(t, err)
+		assert.Equal(t, routeName, route.Name, "route name should be equal")
+		validateGroupVersionKind(t, route)
+	})
+
+	t.Run("get unknown route name returns error", func(t *testing.T) {
```

Comment:
Is there a need for a check for invalid route names?

Reply:
Hi Max, I think it is better for us to cover the invalid failure case and I found we also have this invalid cases for service and revision.

Reply:
No I was thinking about user’s passing a route name like `@#$Q!$#` or anything with invalid HTTP char will be invalid. The API might have a specification as to what route names are valid.

The key for me is not so much that we guard against the valid or invalid name but that we tell user a message that’s easy to understand.

Reply:
Hi @maximilien , got it. I have tried, no matter it is a non-exist or invalid name, kn client can report `routes.serving.knative.dev "xxx" not found` correctly without any http error. such as:

```
jordan at jordandembp.cn.ibm.com in ~/Work/workspace/go/src/github.com/knative/client [master|✔]
✘-1 $ go run cmd/kn/main.go route describe he$low@rld
routes.serving.knative.dev "he@rld" not found
exit status 1

jordan at jordandembp.cn.ibm.com in ~/Work/workspace/go/src/github.com/knative/client [master|✔]
✘-1 $ go run cmd/kn/main.go route describe hello
routes.serving.knative.dev "hello" not found
exit status 1
```

So I also change the test route name to `r@ute-that-d$es-n#t-exist` for spceial case.

Reply:
I think we can rely on the error message from the API client to point in the right direction. I don't think we need an extra validation of the name on our side.